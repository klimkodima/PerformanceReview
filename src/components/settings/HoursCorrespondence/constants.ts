import { t } from 'i18next';
import 'src/i18n';

export const HEADER_CELLS: string[] = [
  t('hours_correspondence.status'),
  t('hours_correspondence.icon'),
  t('hours_correspondence.gap'),
  t('hours_correspondence.confirmation_rate'),
  t('hours_correspondence.tooltip_text')
];

export const ERROR_TEXTS: ErrorTextType = {
  emptyValueError: t('hours_correspondence.empty_value_error'),
  uniqueStatusError: t('hours_correspondence.unique_status_error'),
  confirmationRateError: t('hours_correspondence.confirmation_rate_error'),
  gapMinValueError: t('hours_correspondence.gap_min_value_error'),
  gapMaxValueError: t('hours_correspondence.gap_max_value_error'),
  gapCoverError: t('hours_correspondence.gap_cover_error'),
  gapPatternError: t('hours_correspondence.gap_pattern_error'),
  gapMinMaxValueError: t('hours_correspondence.gap_min_max_value_error')
};

type ErrorTextType = {
  [key: string]: string;
};
