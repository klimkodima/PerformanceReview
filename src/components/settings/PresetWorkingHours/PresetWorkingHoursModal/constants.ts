import { t } from 'i18next';

import { ConstantsTextType } from '../../types';

export const BUTTONS: ConstantsTextType = {
  saveBtn: t('preset_working_hours_table.modal_save_button_text'),
  cancelBtn: t('preset_working_hours_table.modal_cancel_button_text')
};
