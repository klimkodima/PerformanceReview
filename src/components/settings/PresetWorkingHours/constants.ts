import { t } from 'i18next';

import { ConstantsTextType } from '../types';

export const HEADER_CELLS: string[] = [
  t('preset_working_hours_table.header_cell_label'),
  t('preset_working_hours_table.header_cell_hours')
];

export const ERROR_TEXTS: ConstantsTextType = {
  emptyValueError: t('preset_working_hours_table.empty_value_error'),
  uniqueLabelError: t('preset_working_hours_table.unique_label_error'),
  hoursPatternError: t('preset_working_hours_table.hours_pattern_error'),
  hoursValueError: t('preset_working_hours_table.hours_value_error')
};

export const MODAL_ERROR_TEXTS: ConstantsTextType = {
  exceedsValueNotification: t(
    'preset_working_hours_table.modal_notification_exceeds'
  ),
  lessValueNotification: t(
    'preset_working_hours_table.modal_notification_less'
  ),
  emptyValueNotification: t(
    'preset_working_hours_table.modal_notification_empty_table'
  )
};

export const OPTIONS_TEXTS: ConstantsTextType = {
  optionValue1: t('preset_working_hours_table.option_value_1'),
  optionValue2: t('preset_working_hours_table.option_value_2'),
  optionValue3: t('preset_working_hours_table.option_value_3')
};

export const HOURS_PER_DAY = 8;
