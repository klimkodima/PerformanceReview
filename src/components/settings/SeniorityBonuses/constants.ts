import { t } from 'i18next';

export const HEADER_CELLS: string[] = [
  t('seniority_bonuses.header_cell_experience'),
  t('seniority_bonuses.header_cell_bonus')
];

export const CELL_NAMES = {
  experience: t('seniority_bonuses.experience'),
  bonus: t('seniority_bonuses.bonus')
};

export const ERRORS_TEXT = {
  wrongPattern: t('seniority_bonuses.wrong_pattern'),
  emptyField: t('seniority_bonuses.empty_field'),
  intervalError: t('seniority_bonuses.interval_error')
};

export const INFO_TOOLTIP = t('seniority_bonuses.info_tooltip_text');

export const EXPERIENCE_KEY = 'experience';
