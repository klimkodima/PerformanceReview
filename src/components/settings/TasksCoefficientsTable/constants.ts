export const AUDIT_TYPE_MAPPER: ValuesMapperType = {
  initial: 'Initial',
  reaudit: 'Reaudit'
};

type ValuesMapperType = {
  [key: string]: string;
};
