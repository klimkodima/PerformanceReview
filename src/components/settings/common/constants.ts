export const TABLE_BUTTONS: TableBtnsType = {
  editBtn: 'Edit Table',
  addBtn: 'Add',
  deleteBtn: 'Delete',
  saveBtn: 'Save',
  cancelBtn: 'Cancel'
};

type TableBtnsType = {
  [key: string]: string;
};
