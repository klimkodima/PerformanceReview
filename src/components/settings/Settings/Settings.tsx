import { ReactElement, memo } from 'react';

import { SETTINGS_PERMISSION } from 'src/constants';
import { SettingsSidebar } from '../SettingsSidebar';

import { SettingsPermissionType } from 'src/store/App';
import { SettingsTabsType } from '../types';

type SettingsPropsType = {
  listOfTabs: SettingsTabsType[];
  onSettingsTabClick: (name: string) => void;
  settingsPermission: SettingsPermissionType;
};

const Settings = memo(
  ({
    listOfTabs,
    onSettingsTabClick,
    settingsPermission
  }: SettingsPropsType): ReactElement => (
    <>
      <SettingsSidebar
        listOfTabs={listOfTabs}
        onSettingsTabClick={onSettingsTabClick}
      />
      {listOfTabs
        .filter(({ isActive }) => isActive)
        .map(({ name, component, editableComponent }) => (
          <div key={name} className='settings-container'>
            <h2>{name}</h2>
            {settingsPermission === SETTINGS_PERMISSION.WRITE
              ? editableComponent
              : component}
          </div>
        ))}
    </>
  )
);

export default Settings;

Settings.displayName = 'Settings';
