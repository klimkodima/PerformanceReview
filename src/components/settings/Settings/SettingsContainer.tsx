import { ReactElement, useCallback, useState } from 'react';

import { t } from 'i18next';

import { selectSettingsPermission } from 'src/store/App';
import Settings from './Settings';
import { TasksCoefficientsTable } from '../TasksCoefficientsTable';
import {
  HoursCorrespondence,
  HoursCorrespondenceEdit
} from '../HoursCorrespondence';
import { SeniorityBonuses, SeniorityBonusesEdit } from '../SeniorityBonuses';
import {
  PresetWorkingHours,
  PresetWorkingHoursEdit
} from '../PresetWorkingHours';
import { UsersContainer } from '../Users';

import { SettingsTabsType } from '../types';
import { useSelector } from 'react-redux';

export const initListOfTabs: SettingsTabsType[] = [
  {
    name: t('settings.tasks_title'),
    component: <TasksCoefficientsTable />,
    editableComponent: <TasksCoefficientsTable />,
    isActive: false
  },
  {
    name: t('settings.hours_title'),
    component: <HoursCorrespondence />,
    editableComponent: <HoursCorrespondenceEdit />,
    isActive: false
  },
  {
    name: t('settings.preset_working_hours_title'),
    component: <PresetWorkingHours />,
    editableComponent: <PresetWorkingHoursEdit />,
    isActive: false
  },
  {
    name: t('settings.seniority_bonuses_title'),
    component: <SeniorityBonuses />,
    editableComponent: <SeniorityBonusesEdit />,
    isActive: false
  },
  {
    name: t('settings.users_title'),
    component: <UsersContainer />,
    editableComponent: <UsersContainer />,
    isActive: false
  }
];

const SettingsContainer = (): ReactElement => {
  const settingsPermission = useSelector(selectSettingsPermission);

  const [listOfTabs, setListOfTabs] =
    useState<SettingsTabsType[]>(initListOfTabs);

  const handleSettingsTabClick = useCallback((name: string) => {
    setListOfTabs(
      listOfTabs.map((tab) =>
        tab.name === name
          ? { ...tab, isActive: true }
          : { ...tab, isActive: false }
      )
    );
  }, []);

  return (
    <Settings
      settingsPermission={settingsPermission}
      listOfTabs={listOfTabs}
      onSettingsTabClick={handleSettingsTabClick}
    />
  );
};

export default SettingsContainer;
