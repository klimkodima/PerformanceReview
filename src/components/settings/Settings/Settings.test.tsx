import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { I18nextProvider } from 'react-i18next';

import i18n from 'src/i18n';
import store from 'src/store';
import { SETTINGS_PERMISSION } from 'src/constants';
import { initListOfTabs } from './SettingsContainer';
import Settings from './Settings';

describe('Settings', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <I18nextProvider i18n={i18n}>
            <Settings
              settingsPermission={SETTINGS_PERMISSION.WRITE}
              listOfTabs={initListOfTabs}
              onSettingsTabClick={() => true}
            />
          </I18nextProvider>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
