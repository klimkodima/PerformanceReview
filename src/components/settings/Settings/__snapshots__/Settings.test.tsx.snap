// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders correctly 1`] = `
Array [
  <div
    className="settings-sidebar sidebar-wrapper"
  >
    <ul>
      <li
        className="sidebar-list-item active"
        data-testid="sidebar-list-item"
        onClick={[Function]}
      >
        Tasks' Coefficients
      </li>
      <li
        className="sidebar-list-item "
        data-testid="sidebar-list-item"
        onClick={[Function]}
      >
        Hours' Correspondence
      </li>
      <li
        className="sidebar-list-item "
        data-testid="sidebar-list-item"
        onClick={[Function]}
      >
        Preset Working Hours
      </li>
      <li
        className="sidebar-list-item "
        data-testid="sidebar-list-item"
        onClick={[Function]}
      >
        Seniority Bonuses
      </li>
      <li
        className="sidebar-list-item "
        data-testid="sidebar-list-item"
        onClick={[Function]}
      >
        Users
      </li>
    </ul>
  </div>,
  <div
    className="settings-container"
  >
    <h2>
      Tasks' Coefficients
    </h2>
    <div
      className="coeff-table"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root coeff-table-container css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          aria-label="table"
          className="MuiTable-root css-rqglhn-MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium th-main css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Website Group
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium th-main css-1ygcj2i-MuiTableCell-root"
                scope="col"
              />
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium th-main css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Website
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium th-main css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                Audit Type
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
      <div
        className="MuiTablePagination-root coeff-table__paginator css-jtlhu6-MuiTablePagination-root"
      >
        <div
          className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-11bfvty-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            className="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            className="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
            id=":r1:"
          >
            Rows per page:
          </p>
          <div
            className="MuiInputBase-root MuiInputBase-colorPrimary css-16c50h-MuiInputBase-root-MuiTablePagination-select"
            onClick={[Function]}
            variant="standard"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby=":r1: :r0:"
              className="MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              id=":r0:"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              5
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              tabIndex={-1}
              value={5}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            className="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            0â€“0 of 0
          </p>
          <div
            className="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              className="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
